tests/serverpod_test_server/test_integration/test_tools/protocol_test.dart

```diff
@@ -0,0 +1,98 @@
+import 'package:serverpod/serverpod.dart';
+import 'package:test/test.dart';
+
+import 'serverpod_test_tools.dart';
+
+void main() {
+  withServerpod('Given TestToolsEndpoint', (endpoints, session) {
+    test('when calling method with positional arg then echoes the value',
+        () async {
+      final result = await endpoints.testToolsWithArgVariations
+          .echoPositionalArg(session, 'PositionalArg');
+      expect(result, 'PositionalArg');
+    });
+
+    test('when calling method with named arg then echoes the value', () async {
+      final result = await endpoints.testToolsWithArgVariations
+          .echoNamedArg(session, string: 'NamedArg');
+      expect(result, 'NamedArg');
+    });
+
+    test(
+        'when calling method with a nullable named arg with value then echoes value',
+        () async {
+      final result = await endpoints.testToolsWithArgVariations
+          .echoNullableNamedArg(session, string: 'NamedArg');
+      expect(result, 'NamedArg');
+    });
+
+    test(
+        'when calling method with a nullable named arg without passing value then echoes null',
+        () async {
+      final result = await endpoints.testToolsWithArgVariations
+          .echoNullableNamedArg(session);
+      expect(result, isNull);
+    });
+
+    test('when calling method with optional arg then echoes the value',
+        () async {
+      final result = await endpoints.testToolsWithArgVariations
+          .echoOptionalArg(session, 'OptionalArg');
+      expect(result, 'OptionalArg');
+    });
+
+    test('when calling method with optional arg without value then echoes null',
+        () async {
+      final result =
+          await endpoints.testToolsWithArgVariations.echoOptionalArg(session);
+      expect(result, isNull);
+    });
+
+    test('when calling method with positional and named args then echoes args',
+        () async {
+      final result =
+          await endpoints.testToolsWithArgVariations.echoPositionalAndNamedArgs(
```

Comment:
Suggestion(NIT): Add test for when named arg is omitted.

To complete the matrix, this seems to have been missed.

Reply:
45245d4